### DISCUSSION TITLE
Side Panel Enhancements - Actions & Force Focus

### DISCUSSION THREAD

**Discussion by R2bEEaton** [Budibase/budibase#11641] [Ideas]
Just a few side panel related feature requests that I think could be cool.

### Close Actions
- If designers were able to trigger actions when a sidebar was closed, that would be very helpful for things like refreshing data providers or updating state information. This would include triggering when the sidebar closes by losing focus (clicking in the main body of the page) or when it is closed via an action.
- Here's a mock-up. Perhaps the text could "On lose focus" or some better description.
![image](https://github.com/Budibase/budibase/assets/34921506/19f70a89-1069-4092-ac93-0468700b1c02)
- Or, instead of triggering the same action for losing focus or directly closing the sidebar, perhaps there could be two triggers. Another separate action for when the user closes it using a button action or the top right close button, something to distinguish between losing focus and actually completing whatever was in the sidebar.

### Force Focus
- Relatedly, it would also be nice if the sidebar could be forced to stay in focus until some kind of interaction (like a submit button), by disabling interaction with the rest of the page. This could be a toggle in the element's settings.

Let me know what you think!

**Comment by ConorWebb96** [comment]
I like these ideas, I'd recommend creating an enhancement request through issues for this.

I actually ran into the needing the `on close` action earlier this week. Ended up doing what I needed to do on re-opening instead. But this would be a better approach. 🚀
